blog/how-to-virtualize-webassembly-components-with-wasi-virt/index.mdx

```diff
@@ -0,0 +1,170 @@
+
+In this blog, we'll explore how component virtualization works, when you might want to use WASI Virt, and how to get started.
+
+{/* truncate */}
+
+## How component virtualization works
+
+For those new to the WebAssembly component space, a quick review: components are WebAssembly binaries adhering to an additional layer of specification called the [Component Model](https://component-model.bytecodealliance.org/introduction.html). Like ordinary WebAssembly binaries, components are sandboxed and hyper-efficient. The Component Model enables them to go a step further and interoperate&mdash;regardless of the language they were originally written in&mdash;over [standard APIs like the WebAssembly System Interface (WASI)](/docs/concepts/interfaces#well-known-interfaces).
+
+Along with portability and cross-language interoperability, **composability** is a signature feature of components: two or more components can be combined (or "composed") together into one standalone binary, with the constituent components satisfying one another's **dependencies** and **exposed functions**, or "**imports**" and "**exports**" respectively. 
+
+![Animation of components composing together](../../static/img/composed-component.gif)
+
+It's this characteristic that makes it possible to compose a component within an encapsulating component&mdash;in other words, to create a "virtualized component" for which all imports are satisfied, the external requirements abstracted away. In our case, if the requirement we need to abstract away is access to environment variables, we can compose our primary component with an encapsulating component that fulfills the need.
+
+## When to virtualize a component
+
+Perhaps the most important use-case for WASI Virt and component virtualization is creating a sandbox to use sensitive functionality in secure environments. 
+
+In addition to environment variables, WASI Virt can give a component access to a virtualized filesystem, sockets, or stdio. This is an important use-case on wasmCloud, where interfaces like `wasi:sockets` are not available in accordance with a Zero Trust security model. 
+
+Here's the complete list of WASI interfaces that can be used with WASI Virt:
+
+* Clocks
+* Environment
+* Exit
+* Filesystem
+* HTTP
+* Random
+* Sockets
+* Stdio
+
+Beyond requirements for running in secure environments, it's useful to remember that virtualization with WASI Virt is a form of build-time composition, and this makes sense for some use-cases more than others, as well as lending itself more and less ideally to certain development workflows:
```

Comment:
```suggestion
Beyond requirements for running in secure environments, it's useful to remember that virtualization with WASI Virt is a form of [build-time composition](/docs/concepts/linking-components/linking-at-build#example-composition), and this makes sense for some use-cases more than others, as well as lending itself more and less ideally to certain development workflows:
```

Do you think that this would be a valuable link to have? I just saw the keyword